Library/Homebrew/utils.rb

```diff
@@ -245,264 +74,437 @@ def inject_dump_stats!(the_module, pattern)
   # rubocop:enable Style/GlobalVars
 end
 
-def with_homebrew_path
-  with_env(PATH: PATH.new(ENV["HOMEBREW_PATH"])) do
-    yield
+module Kernel
+  def require?(path)
```

Comment:
Can this refactoring go in another PR?

Reply:
Moved to https://github.com/Homebrew/brew/pull/6367.